lib/commands/screenshots.js

```diff
@@ -6,21 +6,26 @@ import log from '../logger';
 let commands = {};
 
 commands.getScreenshot = async function () {
+  const getScreenshotFromWDA = async () => {
+    const data = await this.proxyCommand('/screenshot', 'GET');
+    if (!_.isString(data)) {
+      throw new Error(`Unable to take screenshot. WDA returned '${JSON.stringify(data)}'`);
+    }
+    return data;
+  };
   try {
+    return await getScreenshotFromWDA();
+  } catch (err) {
     if (!this.isRealDevice() && this.xcodeVersion.versionFloat >= 8.1) {
+      log.info(`Falling back to 'simctl io screenshot' API`);
       return await getScreenshot(this.opts.udid);
     }
-  } catch (err) {
-    log.warn(`Cannot make a screenshot using simctl because of "${err.message}". Falling back to WDA API`);
+    let result;
```

Comment:
Ok. I guess. Something about the flow should be in the comments, though. Otherwise this is unclear. Perhaps above line 24 something like `// on real devices retry WDA screenshotting`, so it is obvious that this is a choice.